pkg/detectors/plaidkey/plaidkey.go

```diff
@@ -37,20 +38,22 @@ func (s Scanner) Keywords() []string {
 func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
 	dataStr := string(data)
 
-	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
-	idMatches := idPat.FindAllStringSubmatch(dataStr, -1)
+	// find all the matching keys and ids in the data and make a unique maps for both.
+	uniqueKeys, uniqueIds := make(map[string]struct{}), make(map[string]struct{})
 
-	for _, match := range matches {
-		if len(match) != 2 {
-			continue
-		}
-		resMatch := strings.TrimSpace(match[1])
+	for _, foundKey := range keyPat.FindAllStringSubmatch(dataStr, -1) {
+		uniqueKeys[foundKey[1]] = struct{}{}
+	}
 
-		for _, idMatch := range idMatches {
-			if len(idMatch) != 2 {
-				continue
-			}
-			idresMatch := strings.TrimSpace(idMatch[1])
+	for _, foundId := range idPat.FindAllStringSubmatch(dataStr, -1) {
+		uniqueIds[foundId[1]] = struct{}{}
```

Comment:
Eliminates stuff like `PLAID_CLIENT_ID=XXXXXXXXXXXXXXXXXXXXXXXX`.
```suggestion
		id := foundId[1]
		if detectors.StringShannonEntropy(id) < 3 {
			continue
		}

		uniqueIds[id] = struct{}{}
```

Reply:
Thanks @rgmz! I didn't think of this, but any detectors I may work on later I'll definitely add this!